picasso/src/main/java/com/squareup/picasso3/ImageDecoderFactory.java

```diff
@@ -0,0 +1,28 @@
+package com.squareup.picasso3;
+
+import androidx.annotation.Nullable;
+import java.util.List;
+import okio.BufferedSource;
+
+final class ImageDecoderFactory {
+
+  final List<ImageDecoder> decoders;
+
+  ImageDecoderFactory(List<ImageDecoder> decoders) {
+    this.decoders = decoders;
+  }
+
+  /**
+   * Returns the first {@link ImageDecoder} that can handle the supplied <code>source</code>.
+   * @param source The source of the image data.
+   * @return The first ImageDecoder that can decode the source, or null.
+   */
+  @Nullable ImageDecoder getImageDecoderForSource(BufferedSource source) {
+    for (ImageDecoder decoder : decoders) {
```

Comment:
nit: fori loop?

Reply:
Can you elaborate on the suggestion? Fori?

Reply:
`for (i=0...` to avoid the iterator in the enhanced for loop. not much of a big deal nowadays.

Reply:
Fixed in c2fa402